### PR TITLE
Refactor library structure

### PR DESCRIPTION
**Pull request by OptimumCode**

### CODE REVIEW

File: pkg/factory/configProvider.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	}
+}
+
+type fileConfigProvider struct {
+	configFS      fs.FS
+	fileExtension string
+	zLogger       *zerolog.Logger
+}
+
+func (cfd *fileConfigProvider) GetConfig(resourceName string, target interface{}) error {
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/factory/configProvider.go:53]:
```suggestion
func (cfd *fileConfigProvider) GetConfig(resourceName string, target any) error {
```
Could we use any type alias here?

File: pkg/metrics/commonMetrics.go [hunk 2]
```diff
@@ -21,33 +20,33 @@ import (
...
+	DefaultExchangeLabelName     = "exchange"
+	DefaultRoutingKeyLabelName   = "routing_key"
+	DefaultQueueLabelName        = "queue"
+	DefaultMessageTypeLabelName  = "message_type"
+	DefaultTh2PinLabelName       = "th2_pin"
+	DefaultTh2TypeLabelName      = "th2_type"
+	MessageGroupTh2Type          = "MESSAGE_GROUP"
+	EventTh2Type                 = "EVENT"
+	RawMessageType               = "RAW_MESSAGE"
+	ParsedMessageType            = "MESSAGE"
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/metrics/commonMetrics.go:37]: For mine information. What is the correct naming convention.
I found two opinions: 
https://www.golangprograms.com/go-language/constants.html
https://go.dev/doc/effective_go#mixed-caps

File: pkg/queue/event/sender.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package event
+
+import p_buff "th2-grpc/th2_grpc_common"
+
+type BatchSender interface {
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/queue/event/sender.go:20]: May BatchSender and BatchSubscriber should be private interfaces

File: pkg/queue/message/subscriber.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package message
+
+type GroupBatchSubscriber interface {
+	SetListener(listener Listener)
+	RemoveListener()
+	//close()
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/queue/message/subscriber.go:21]:
```suggestion
```

File: pkg/queue/rabbitmq/connection/connectionManager.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+type Manager struct {
+	Publisher *Publisher
+	Consumer  *Consumer
+
+	Logger zerolog.Logger
+}
+
+func NewConnectionManager(connConfiguration Config, logger zerolog.Logger) (Manager, error) {
```
Comment (Nikita-Smirnov-Exactpro) [inline pkg/queue/rabbitmq/connection/connectionManager.go:31]: Maybe ConnectionManager should be private?

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

File: pkg/factory/configProvider.go (hunk 1 above)
Comment (OptimumOpium) [reply pkg/factory/configProvider.go:53]: Yes, we can. Missed that place

File: pkg/metrics/commonMetrics.go (hunk 2 above)
Comment (OptimumOpium) [reply pkg/metrics/commonMetrics.go:37]: I would believe the official guide from go.dev. Also, IDE for golang reports the original formatting

File: pkg/queue/event/sender.go (hunk 3 above)
Comment (OptimumOpium) [reply pkg/queue/event/sender.go:20]: We don't even need them. I will remove those interfaces for messages and events

File: pkg/queue/rabbitmq/connection/connectionManager.go (hunk 5 above)
Comment (OptimumOpium) [reply pkg/queue/rabbitmq/connection/connectionManager.go]: Let's do it. It is better to expose it in future if we need it

**Review by Nikita-Smirnov-Exactpro — DISMISSED** [review]

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Pull request merged**